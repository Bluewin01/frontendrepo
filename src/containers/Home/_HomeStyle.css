* {
  font-size: 20px;
}

.body {
  display: flex;
  min-height: 100vh;
}

.content {
  flex-grow: 1;
  padding: 0px 10px 10px 10px;
  background-color: #f1efef;
}

label {
  margin-top: 15px;
  font-weight: 500;
}

.templateCategory {
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(180px, 1fr));
  grid-gap: 0.5rem;
  grid-row-gap: 0px;
  flex-wrap: wrap;
}

.background {
  margin-top: 50px;
  position: absolute;
  text-align: center;
  margin-left: 350px;
}
