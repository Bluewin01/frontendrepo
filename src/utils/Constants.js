export const theme = {
  color: {
    w: "#FFFFFF",
    k: "#000000",
    p1: "#D31145",
    p2: "#596C80",
    p3: "#554344",
    p4: "#363E3F",
    p5: "#e6edeb",
    s1: "#E86487",
    s2: "#FBBF56",
    s3: "#97CB5D",
    s4: "#46D3BD",
    s5: "#22A8DA",
    s6: "#9962D2",
    t1: "#F6CCD7",
    t2: "#B8123E",
    t3: "#F9E1B8",
    t4: "#C36418",
    t5: "#C6EB9E",
    t6: "#65962E",
    b1: "#D4EDF1",
    b2: "#FAF9F3",
    b3: "#F0EEE1",
    b4: "#DBDAD4",
    b5: "#B3B1A8",
    b6: "#6F6E68",
    b8: "#302F2C",
    b9: "#292025",
    u2: "#FF5500",
    u3: "#405B70",
    u4: "#769FAB",
  },
};

export const BASE_API = `${process.env.REACT_APP_BASE_API_URL}`;
