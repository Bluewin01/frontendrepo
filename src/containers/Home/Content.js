import React, { useState, useEffect } from "react";
import "./_HomeStyle.css";
import History from "../../routes/History";
import sampleTemplateList from "../../assets/sampleTemplate/sampleTemplate.json";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import {
  makeSelectTemplateList,
  makeSelectTemplateType,
} from "../../store/Template/TemplateReselect";
import PropTypes from "prop-types";
import { Card } from "../../components/Card/Card";
import * as TemplateActions from "../../store/Template/TemplateActions";
import { Modal } from "../../components/Modal/Modal";
import "react-toastify/dist/ReactToastify.css";

const Content = (props) => {
  const {
    templateType,
    templateList,
    initiateViewTemplate,
    initiateDeleteTemplate,
  } = props;
  const sampleTemplate = sampleTemplateList.filter((sample) => {
    return sample.type === templateType;
  });
  const [showDeleteModal, setShowDeleteModal] = useState(false);
  const [selectedTemplateId, setSelectedTemplateId] = useState("");

  const createTemplate = () => {
    if (templateType === "Email") {
      History.push("/email");
    }
    if (templateType === "Sms") {
      History.push("/sms");
    }
    if (templateType === "Push") {
      History.push("/push");
    }
  };

  const editTemplate = (templateId) => {
    if (templateType === "Email") {
      initiateViewTemplate(templateId);
      History.push("/email");
    }
    if (templateType === "Sms") {
      initiateViewTemplate(templateId);
      History.push("/sms");
    }
    if (templateType === "Push") {
      initiateViewTemplate(templateId);
      History.push("/push");
    }
  };

  const deleteTemplate = () => {
    initiateDeleteTemplate(selectedTemplateId, templateType);
    setShowDeleteModal(false);
  };

  return (
    <section className="content">
      <label>Sample Templates</label>
      <div className="templateCategory">
        {sampleTemplate.map((sample) => (
          <Card
            key={sample.id}
            cardType={"templateList"}
            template={sample}
            deleteTemplate={deleteTemplate}
            buttons={[
              {
                name: "Use Template",
                onClick: () => alert("This feature is not available yet"),
              },
            ]}
          />
        ))}
      </div>
      <label>My Templates</label>
      <div className="templateCategory">
        <Card cardType={"createTemplate"} createTemplate={createTemplate} />
        {templateList.map((template) => (
          <Card
            key={template.id}
            cardType={"templateList"}
            template={template}
            buttons={[
              { name: "Edit", onClick: editTemplate },
              {
                name: "Delete",
                onClick: () => {
                  setShowDeleteModal(true);
                  setSelectedTemplateId(template.id);
                },
              },
            ]}
          />
        ))}
      </div>
      {showDeleteModal === true ? (
        <Modal
          headerTitle={`Are you sure to delete ${templateType} template?`}
          showBody={false}
          buttons={[
            {
              name: "Confirm",
              color: "danger",
              handler: deleteTemplate,
            },
            {
              name: "Cancel",
              color: "light",
              handler: () => setShowDeleteModal(false),
            },
          ]}
        />
      ) : null}
    </section>
  );
};

const mapStateToProps = createStructuredSelector({
  templateType: makeSelectTemplateType(),
  templateList: makeSelectTemplateList(),
});

const mapDispatchToProps = (dispatch) => {
  return {
    initiateViewTemplate: (templateId) =>
      dispatch(
        TemplateActions.viewTemplateRequest({
          templateId,
        })
      ),
    initiateDeleteTemplate: (templateId, templateType) =>
      dispatch(
        TemplateActions.deleteTemplateRequest({
          templateId,
          templateType,
        })
      ),
  };
};

Content.propTypes = {
  templateType: PropTypes.string,
  templateList: PropTypes.array,
  initiateViewTemplate: PropTypes.func,
  initiateDeleteTemplate: PropTypes.func,
};

export default connect(mapStateToProps, mapDispatchToProps)(Content);
