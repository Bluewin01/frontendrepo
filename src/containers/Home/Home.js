import React, { useEffect } from "react";
import "./_HomeStyle.css";
import Content from "./Content";
import * as TemplateActions from "../../store/Template/TemplateActions";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import {
  makeSelectTemplateType,
  makeSelectError,
  makeSelectResponse,
} from "../../store/Template/TemplateReselect";
import PropTypes from "prop-types";
import { ToastContainer, toast } from "react-toastify";
import "react-toastify/dist/ReactToastify.css";
import Sidebar from "../../components/Sidebar/Sidebar";
import AIALogo from "./../../assets/images/aiawhite-logo.png";

const Home = (props) => {
  const {
    templateType,
    initiateGetTemplateList,
    error,
    response,
    setTemplateObjectToEmpty,
    initiateSetResponseToDefault,
    initiateSetErrorToDefault,
  } = props;

  useEffect(() => {
    initiateGetTemplateList(templateType);
    setTemplateObjectToEmpty();
    window.scrollTo(0, 0);
  }, []);

  useEffect(() => {
    if (error) toast.error(error);
    if (response) toast.success(response);
    initiateSetErrorToDefault();
    initiateSetResponseToDefault();
  }, [error, response]);

  const handleClick = (menu) => {
    initiateGetTemplateList(menu);
  };

  return (
    <>
      <ToastContainer
        position="top-right"
        autoClose={5000}
        hideProgressBar
        newestOnTop={false}
        closeOnClick
        rtl={false}
        pauseOnFocusLoss
        draggable
        pauseOnHover={false}
      />
      <section className="body">
        <Sidebar templateType={templateType} handleClick={handleClick} />
        <div className="content">
          <div className="text-center background">
            <img src={AIALogo} alt="aia-logo-white" />
          </div>
          <Content />
        </div>
      </section>
    </>
  );
};

const mapStateToProps = createStructuredSelector({
  templateType: makeSelectTemplateType(),
  error: makeSelectError(),
  response: makeSelectResponse(),
});

const mapDispatchToProps = (dispatch) => {
  return {
    initiateGetTemplateList: (templateType) =>
      dispatch(TemplateActions.getTemplateListRequest({ templateType })),
    setTemplateObjectToEmpty: () =>
      dispatch(TemplateActions.setTemplateObjectToEmpty()),
    initiateSetResponseToDefault: () =>
      dispatch(TemplateActions.setResponseToDefault()),
    initiateSetErrorToDefault: () =>
      dispatch(TemplateActions.setErrorToDefault()),
  };
};

Content.propTypes = {
  initiateGetTemplateList: PropTypes.func,
  username: PropTypes.string,
  templateType: PropTypes.string,
  error: PropTypes.string,
  setTemplateObjectToEmpty: PropTypes.func,
  initiateSetResponseToDefault: PropTypes.func,
  initiateSetErrorToDefault: PropTypes.func,
};

export default connect(mapStateToProps, mapDispatchToProps)(Home);
