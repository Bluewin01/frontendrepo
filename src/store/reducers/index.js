import { combineReducers } from "redux";
import { configureStore } from "@reduxjs/toolkit";
import lang from "../Lang/LangReducer";
import loader from "../Loader/LoaderReducer";
import Auth from "../Auth/AuthReducer";
import Template from "../Template/TemplateReducer";

const rootReducer = combineReducers({
  lang,
  loader,
  Auth,
  Template,
});

export const setupStore = (preloadedState) => {
  return configureStore({
    reducer: rootReducer,
    preloadedState,
  });
};

export default rootReducer;
