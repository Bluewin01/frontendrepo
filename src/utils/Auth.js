// Service to check authentication for user and to signOut
import * as AuthActions from "../store/Auth/AuthActions";
import store from "../store";

const Auth = {
  signOut() {
    store.dispatch(AuthActions.signOut);
  },
  isAuth(jwt) {
    if (!jwt) {
      return false;
    } else {
      return true;
    }
  },
};
export default Auth;
