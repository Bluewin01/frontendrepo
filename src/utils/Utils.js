// test-utils.jsx
import React from "react";
import { render as rtlRender } from "@testing-library/react";
import { configureStore } from "@reduxjs/toolkit";
import { Provider } from "react-redux";
import { setupStore } from "../store/reducers";

export function render(
  ui,
  { preloadedState, store = setupStore(preloadedState), ...renderOptions } = {}
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }

  // Return an object with the store and all of RTL's query functions
  return { store, ...rtlRender(ui, { wrapper: Wrapper, ...renderOptions }) };
}

export * from "@testing-library/react";
