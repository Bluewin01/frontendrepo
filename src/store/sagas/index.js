import { AuthSaga } from "../Auth/AuthSaga";
import { TemplateSaga } from "../Template/TemplateSaga";
import { fork, all } from "redux-saga/effects";

export function* watchSagas() {
  yield all([AuthSaga(), TemplateSaga()]);
}
