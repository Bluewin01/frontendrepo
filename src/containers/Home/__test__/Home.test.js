import React from "react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import Home from "../Home";
import store from "../../../store";
import { Provider } from "react-redux";
import { render, fireEvent, screen, act } from "../../../utils/Utils";

describe("Containers: Home", () => {
  beforeAll(() => {
    Object.defineProperty(window, "matchMedia", {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // Deprecated
        removeListener: jest.fn(), // Deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  });
  it("should be render correctly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Home />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should be shown My Email Template when render home page with templateType: Email", () => {
    render(<Home />, {
      preloadedState: {
        Template: {
          templateType: "Email",
          error: null,
          response: null,
          templateList: [
            {
              id: "fakeId",
              name: "fakeName",
              _id: "fake_id",
            },
          ],
        },
      },
    });

    expect(screen.getByText("My Email Template")).toBeInTheDocument();
  });

  it("should be shown My Sms Template when render home page with templateType: Sms", async () => {
    render(<Home />, {
      preloadedState: {
        Template: {
          templateType: "Sms",
          error: null,
          response: null,
          templateList: [
            {
              id: "fakeId",
              name: "fakeName",
              _id: "fake_id",
            },
          ],
        },
      },
    });

    const smsSidebar = screen.getByText("Sms");

    act(() => {
      fireEvent.click(smsSidebar);
    });

    expect(screen.getByText("My Sms Template")).toBeInTheDocument();
  });

  it("should be shown My Push Template when render home page with templateType: Push", () => {
    render(<Home />, {
      preloadedState: {
        Template: {
          templateType: "Push",
          error: null,
          response: null,
          templateList: [
            {
              id: "fakeId",
              name: "fakeName",
              _id: "fake_id",
            },
          ],
        },
      },
    });

    const pushSidebar = screen.getByText("Push");

    act(() => {
      fireEvent.click(pushSidebar);
    });

    expect(screen.getByText("My Push Template")).toBeInTheDocument();
  });

  it("should be shown toast error message when render home page with error", async () => {
    render(<Home />, {
      preloadedState: {
        Template: {
          templateType: "Email",
          error: "fakeError",
          response: null,
          templateList: [
            {
              id: "fakeId",
              name: "fakeName",
              _id: "fake_id",
            },
          ],
        },
      },
    });

    expect(await screen.findByText("fakeError")).toBeInTheDocument();
  });

  it("should be shown toast success message when render home page with response", async () => {
    render(<Home />, {
      preloadedState: {
        Template: {
          templateType: "Email",
          error: null,
          response: "fakeResponse",
          templateList: [
            {
              id: "fakeId",
              name: "fakeName",
              _id: "fake_id",
            },
          ],
        },
      },
    });

    expect(await screen.findByText("fakeResponse")).toBeInTheDocument();
  });
});
