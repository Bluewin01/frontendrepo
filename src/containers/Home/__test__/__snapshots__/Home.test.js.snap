// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Containers: Home should be render correctly 1`] = `
Array [
  <div
    className="Toastify"
  />,
  <section
    className="body"
  >
    <div
      className="sc-bdVaJa efNFrH"
    >
      <div
        className="sc-bwzfXH kaXCHS"
      >
        <button
          className="sc-htpNat gSFzgt"
          onClick={[Function]}
        >
          <div>
            <svg
              fill="currentColor"
              height={28}
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width={28}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 14H4V8l8 5 8-5v10zm-8-7L4 6h16l-8 5z"
              />
            </svg>
          </div>
          <div>
            Email
          </div>
        </button>
        <button
          className="sc-htpNat gJvYaN"
          onClick={[Function]}
        >
          <div>
            <svg
              fill="currentColor"
              height={28}
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width={28}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M20 2H4c-1.1 0-2 .9-2 2v18l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm0 14H5.17L4 17.17V4h16v12zM7 9h2v2H7zm8 0h2v2h-2zm-4 0h2v2h-2z"
              />
            </svg>
          </div>
          <div>
            Sms
          </div>
        </button>
        <button
          className="sc-htpNat gJvYaN"
          onClick={[Function]}
        >
          <div>
            <svg
              fill="currentColor"
              height={28}
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width={28}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M18 16v-5c0-3.07-1.64-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.63 5.36 6 7.92 6 11v5l-2 2v1h16v-1l-2-2zm-5 0h-2v-2h2v2zm0-4h-2V8h2v4zm-1 10c1.1 0 2-.9 2-2h-4a2 2 0 002 2z"
              />
            </svg>
          </div>
          <div>
            Push
          </div>
        </button>
        <div
          className="sc-bxivhb fJVlpn"
          type="Email"
        />
      </div>
    </div>
    <div
      className="content"
    >
      <div
        className="text-center background"
      >
        <img
          alt="aia-logo-white"
          src={Object {}}
        />
      </div>
      <section
        className="content"
      >
        <label>
          Sample Templates
        </label>
        <div
          className="templateCategory"
        >
          <div
            className="card"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="cardContent"
              style={
                Object {
                  "display": "block",
                }
              }
            >
              My Email Template
            </div>
            <div
              className="hiddenButton"
              style={
                Object {
                  "display": "none",
                }
              }
            >
              <div
                className="cardButton"
                data-testid="editTemplate"
                onClick={[Function]}
                style={
                  Object {
                    "height": "160px",
                  }
                }
              >
                Use Template
              </div>
            </div>
          </div>
        </div>
        <label>
          My Templates
        </label>
        <div
          className="templateCategory"
        >
          <div
            className="card"
            data-testid="createTemplate"
            onClick={[Function]}
          >
            <div
              className="cardContent"
            >
              <span
                aria-label="plus"
                className="anticon anticon-plus iconDesign"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="plus"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <defs>
                    <style />
                  </defs>
                  <path
                    d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
                  />
                  <path
                    d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
                  />
                </svg>
              </span>
            </div>
          </div>
        </div>
      </section>
    </div>
  </section>,
]
`;
